blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/AbstractPipelineImpl.java

```diff
@@ -206,6 +212,20 @@ public Resource resolve(Item context, Reachable parent, Item target) {
     }
 
     @Override
+    public int getNumberOfRunningPipelines(){
+        RunList runningJobs =  job.getBuilds().filter(isRunning);
+        return Iterators.size(runningJobs.iterator());
+    }
+
+    @Override
+    public int getNumberOfQueuedPipelines() {
+        if(job instanceof BuildableItem) {
```

Comment:
@kzantow no, I don't think so. BuildableItem can be scheduled and queued so thats right contract to check against.